tests/libs/block_generator/block_generator_tests.cpp

```diff
@@ -0,0 +1,181 @@
+#include <gtest/gtest.h>
+
+#include <boost/json.hpp>
+#include <boost/json/stream_parser.hpp>
+#include <cstdint>
+#include <fstream>
+#include <initializer_list>
+
+#include "block_generator.hpp"
+#include "zkevm_framework/data_types/block.hpp"
+#include "zkevm_framework/data_types/transaction.hpp"
+
+namespace data_types::block_generator::test {
+
+    std::optional<boost::json::value> parse_test_file(const char *path) {
+        std::ifstream input_file(path);
+        if (!input_file.is_open()) {
+            return {};
+        }
+        return parse_json(input_file);
+    }
+
+    template<typename T>
+    bool compare_bytes(const T &byte_container, std::initializer_list<uint8_t> expected) {
+        if (byte_container.size() != expected.size()) {
+            std::cout << "sizes" << std::endl;
+            return false;
+        }
+        for (auto it = expected.begin(); it != expected.end(); ++it) {
+            if (static_cast<typename T::value_type>(*it) != byte_container[it - expected.begin()]) {
+                std::cout << std::hex << (int)*it << std::endl;
+                return false;
+            }
+        }
+        return true;
+    }
+
+    TEST(generator_test, smoke_test) {
+        // Simply check if the format description was parsed without any errors
+        std::optional<boost::json::value> jvalue =
+            parse_test_file(INPUT_PATH "/inputs/scheme.json");
```

Comment:
we really need the separate file for check format parsing?

Reply:
It could be useful to make sure if our documentation is correct.